### PR TITLE
[TTAHUB-656] Goal Close/Suspend Reason Modal

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

If a user attempts to Close or Suspend a Goal we need them to provide a reason.

## How to test

Go to any recipient and view the 'Goals & Objectives' tab. When attempting to Close or Suspend a Goal you should be prompted to fill out a model with a reason and optional context.

**Note:** Before we push this PR the code to change the actual Goal status and reason will be _commented out_.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-656

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
~- [ ] API Documentation updated~
~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Review by jasalisbury — APPROVED** [review]
:+1:

File: frontend/src/components/CloseSuspendReasonModal.css [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.smart-hub--goal-close-suspend-reason .usa-form-group fieldset textarea {
+    resize: none;
+    width: 415px;
```
Comment (thewatermethod) [inline frontend/src/components/CloseSuspendReasonModal.css:3]: why turn off resize?

File: frontend/src/components/CloseSuspendReasonModal.css [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+.usa-modal__footer .usa-button-group__item .usa-button--unstyled {
+    padding-left: 8px;
+    padding-right: 8px;
+    margin-right: 8px;
+    margin-left: 8px;
+}
```
Comment (thewatermethod) [inline frontend/src/components/CloseSuspendReasonModal.css:11]: functionally the same as padding-x-1 margin-x-1, can we use the usdws classes instead?

File: frontend/src/components/CloseSuspendReasonModal.css [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+
+.popup-modal .usa-modal__main h2 {
+    margin-bottom: 0px;
```
Comment (thewatermethod) [inline frontend/src/components/CloseSuspendReasonModal.css:15]: This and some of the other CSS in this file apply to all modals, should it go in a different file, maybe Modal.css?

File: frontend/src/components/CloseSuspendReasonModal.css (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/CloseSuspendReasonModal.css:3]: Why not.

File: frontend/src/components/CloseSuspendReasonModal.css (hunk 3 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/CloseSuspendReasonModal.css:15]: Its not for all modals just this one.

File: frontend/src/components/CloseSuspendReasonModal.css (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/CloseSuspendReasonModal.css:11]: Whats being selected here is in the base modal class. We only want this change for the close/suspend modal. 

Let me know if you think there is a better way to do this in CSS.

File: frontend/src/components/CloseSuspendReasonModal.css (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/CloseSuspendReasonModal.css:3]: In all seriousness, its fixed in the mockup.

File: frontend/src/components/CloseSuspendReasonModal.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  const [showValidationError, setShowValidationError] = useState(false);
+
+  useEffect(() => {
+    // Every time we show the modal reset the form.
+    setCloseSuspendReason('');
+    setCloseSuspendContext('');
+    setShowValidationError(false);
+  }, [resetValues]);
+
+  const reasonDisplayStatus = newStatus === 'Completed' ? 'closing' : 'suspending';
```
[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply frontend/src/components/CloseSuspendReasonModal.js:25]: Not sure I follow here. You want 'closing' and 'suspending' in constants? These are only used for the modal title.

File: frontend/src/components/GoalsTable/GoalRow.js [hunk 5]
```diff
@@ -39,6 +40,19 @@ function GoalRow({
...
+  // Close/Suspend Reason Modal.
+  const [closeSuspendGoalId, setCloseSuspendGoalId] = useState(0);
+  const [closeSuspendStatus, setCloseSuspendStatus] = useState('');
+  const [resetModalValues, setResetModalValues] = useState(false);
+  const closeSuspendModalRef = useRef();
+
+  const showCloseSuspendGoalModal = (status, goalId) => {
+    setCloseSuspendGoalId(goalId);
+    setCloseSuspendStatus(status);
+    setResetModalValues(!resetModalValues); // Always flip to trigger form reset useEffect.
```
[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply frontend/src/components/GoalsTable/GoalRow.js:52]: Not sure how the previous state would determine the new state. We always want to make sure the modal form is clear when open (regardless of before).

Let me know if I'm misunderstanding.

File: frontend/src/components/CloseSuspendReasonModal.css (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/CloseSuspendReasonModal.css:11]: Added close/suspend modal prefix in css.

File: frontend/src/components/CloseSuspendReasonModal.css (hunk 3 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/CloseSuspendReasonModal.css:15]: Added close/suspend modal prefix in css.

File: frontend/src/components/CloseSuspendReasonModal.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  const reasonRadioOptions = newStatus === 'Completed' ? GOAL_CLOSE_REASONS : GOAL_SUSPEND_REASONS;
+  const ReasonChanged = (e) => {
+    setCloseSuspendReason(e.target.value);
+    setShowValidationError(false);
+  };
+  const generateReasonRadioButtons = () => reasonRadioOptions.map((r) => (
+    <Radio
+      id={`${goalId}-${r}-radio-reason`}
+      key={`${goalId}-${r}-radio-reason`}
```
Comment (thewatermethod) [inline frontend/src/components/CloseSuspendReasonModal.js:34]: HTML ids must start with a letter, and they can't have spaces, please adjust line 33 accordingly

**Review by thewatermethod — CHANGES_REQUESTED** [review]
see comment

**Pull request closed without merging**